src/audio/dai-zephyr.c

```diff
@@ -1322,15 +1317,23 @@ static int dai_copy(struct comp_dev *dev)
 
 	ret = dma_reload(dd->chan->dma->z_dev, dd->chan->index, 0, 0, copy_bytes);
 	if (ret < 0) {
-		dai_report_xrun(dev, copy_bytes);
+		dai_report_xrun(dd, dev, copy_bytes);
 		return ret;
 	}
 
-	dai_dma_position_update(dev);
+	dai_dma_position_update(dd, dev);
 
 	return ret;
 }
 
+/* copy and process stream data from source to sink buffers */
+static int dai_copy(struct comp_dev *dev)
```

Comment:
this could be an inline in the header.

Reply:
it is part of drv interface, and copier actually does not use dai_copy, it is using dai_zephyr_copy.